src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -2269,6 +2278,57 @@ private static bool EmitComparison(BinaryExpression expr, IList<ParameterExpress
                 return false;
             }
 
+            private static bool EmitArithmeticOperation(BinaryExpression expr, IList<ParameterExpression> ps, ILGenerator il, ClosureInfo closure)
+            {
+                var leftExpr = expr.Left;
+                var rightExpr = expr.Right;
+                if (!leftExpr.Type.GetTypeInfo().IsPrimitive)
+                {
+                    //TODO: implement arithmetic operations for non primitive classes and structs (like Decimal).
+                    return false;
+                }
+                if (!TryEmit(leftExpr, leftExpr.NodeType, leftExpr.Type, ps, il, closure) ||
+                    !TryEmit(rightExpr, rightExpr.NodeType, rightExpr.Type, ps, il, closure))
+                    return false;
+
+                bool isUnsigned = leftExpr.Type == typeof(ushort) ||
+                                  leftExpr.Type == typeof(uint) ||
+                                  leftExpr.Type == typeof(ulong);
```

Comment:
General question, should we include `byte` as well?

Reply:
Wow, good catch.
Thanks for a good comment.